### ISSUE TITLE
Issue when running with happypack

### ISSUE THREAD

**Issue by v17al** [mzgoddard/hard-source-webpack-plugin#326]
On the latest NPM version of hard-source-webpack-plugin (including the alphas) and happypack, I get the following once hard-source-webpack-plugin is added to my list of plugins (works fine without it):
```
Happy[ts]: Version: 5.0.0. Threads: 2
<<project location>>\node_modules\webpack\node_modules\enhanced-resolve\lib\Resolver.js:32
                throw new Error("Signature changed: context parameter added");
                ^

Error: Signature changed: context parameter added
    at Resolver.resolve (<<project location>>\node_modules\webpack\node_modules\enhanced-resolve\lib\Resolver.js:32:9)
    at Resolver.resolver.resolve (<<project location>>\node_modules\hard-source-webpack-plugin\lib\enhanced-resolve-cache.js:329:22)
    at resolveLoader (<<project location>>\node_modules\happypack\lib\WebpackUtils.js:150:17)
    at <<project location>>\node_modules\happypack\lib\WebpackUtils.js:131:7
    at <<project location>>\node_modules\happypack\node_modules\async\lib\async.js:713:13
    at async.forEachOf.async.eachOf (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:233:13)
    at _parallel (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:712:9)
    at Object.async.parallel (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:726:9)
    at Object.exports.resolveLoaders (<<project location>>\node_modules\happypack\lib\WebpackUtils.js:127:9)
    at resolveLoaders (<<project location>>\node_modules\happypack\lib\HappyPlugin.js:139:20)
    at <<project location>>\node_modules\happypack\node_modules\async\lib\async.js:713:13
    at iterate (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:262:13)
    at async.forEachOfSeries.async.eachOfSeries (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:281:9)
    at _parallel (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:712:9)
    at Object.async.series (<<project location>>\node_modules\happypack\node_modules\async\lib\async.js:734:9)
    at HappyPlugin.start (<<project location>>\node_modules\happypack\lib\HappyPlugin.js:128:9)

npm ERR! Windows_NT 10.0.16299
```

Any ideas?

**Comment by mzgoddard** [comment]
@v17al I need to know what plugins and loaders you are using. I made a quick typescript-hardsource-happypack project to see if I could reproduce this but couldn't. If you could make a small project that reproduces this error that will help immensely.

**Comment by mdentremont** [comment]
Duplicate of https://github.com/mzgoddard/hard-source-webpack-plugin/issues/380

**Comment by v17al** [comment]
Duplicate of #380

**Comment by budlion** [comment]
@v17al  @mzgoddard  Can you reopen this Issue. I've got the same Issue.

webpack: 3.12.0
enhanced-resolve: 3.4.1
happypack: 4.0.1 || 5.0.1

enhanced-resolve 3.x does not support 3 arguments, while hard-source-webpack-plugin will trap the code that try to give resolve 3 arguments.

```
// enhanced-resolve
Resolver.prototype.resolve = function resolve(context, path, request, callback) {
	if(arguments.length === 3) {
		throw new Error("Signature changed: context parameter added" + arguments[0] + ' ' + arguments[1] + ' ' + arguments[2]);
	}
}
```
```
// hard-source-webpack-plugin CacheEnhancedResolve.sj

          if (numArgs === 3) { 
            _resolve.call(this, context, request, callback); // will trap here
          } else if (numArgs === 5) {
            _resolve.call(
              this,
              info,
              context,
              request,
              resolveContext,
              callback,
            );
          } else {
            _resolve.call(this, info, context, request, callback);
          }

```
```
// happypack
// Ops! our resolver has 5 arguments ...
  function resolveLoader(context, loader, callback) {
    if (compiler.resolverFactory) {
      compiler.resolverFactory.get("loader").resolve({}, context, loader, {}, callback);
    } else {
      var resolve = compiler.resolvers.loader.resolve;
      var resolveContext = compiler.resolvers.loader;

      // webpack2 has changed the signature for the resolve method where it accepts
      // a fourth argument (context), so we need to sniff and support both versions
      //
      // fixes #23
      if (resolve.length === 4) {
        resolve.apply(resolveContext, [ context, context, loader, callback ])
      }
      else {
        resolve.apply(resolveContext, [ context, loader, callback ]);
      }
    }
  }
```
